src/request/containers/RequestDetailPanelDataOperationCommandContainer.ts

```diff
@@ -0,0 +1,33 @@
+import { getSelectedOperation } from '../selectors/RequestDetailDataSelectors';
+import { IRequestState } from '../stores/IRequestState';
+import { RequestDetailPanelDataOperationCommand, IRequestDetailPanelDataOperationCommandProps } from '../components/RequestDetailPanelDataOperationCommand';
+
+import { connect } from 'react-redux';
+
+function getLanguageForDatabase(database: string): string {
+    switch (database) {
+        case 'MongoDB':
+            return 'json';
+        default: 
+            return 'sql';
+    }
+}
+
+function mapStateToProps(state: IRequestState, ownProps): IRequestDetailPanelDataOperationCommandProps {
+    const operation = getSelectedOperation(state);
+
+    return {
+        command: operation ? operation.command : '',
+        language: getLanguageForDatabase(operation ? operation.database : undefined)
+    }    
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+    };
+}
+
+export const RequestDetailPanelDataOperationCommandContainer = connect(
+    mapStateToProps,
+    mapDispatchToProps
+)(RequestDetailPanelDataOperationCommand);
```

Comment:
Also wondering if this `mapStateToProps` should happen here, instead of within `RequestDetailPanelDataContainer`? Given that `RequestDetailPanelDataOperationCommandContainer` is now "smart" if this is ideal or not. I'm assuming you have seen https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.xvswxbfmp.